fix(anthropic): handle orphaned tool calls and improve error propagation

mpj opened the pull request:
### Describe your changes

This PR fixes an session-breaking issue where Anthropic API calls would fail with "tool_use ids were found without tool_result blocks" error when the assistant's response was interrupted or when there were issues with tool execution.

This issue surfaced during Anthropic's service outage on August 21, 2025, revealing a edge case in how Crush handles interrupted API interactions. During the outage, users experienced network interruptions mid-conversation, particularly while tool calls were being executed. 

This created a bunch of errors that I don't normally see under normal conditions thus hard to replicate.  The loading spinner continues indefinitely, giving no indication of the actual error and the error wasn't properly propagated through the event stream, leaving the UI stuck in a "Working..." state with no indication that it had frozen, and sometimes this would also lead to errors when trying to resume (or even summarize) the chat after restarting crush. 

### The Problem in Detail

When examining the logs and database, we discovered that conversations interrupted during tool
execution left the message history in an inconsistent state:

```sql
-- Database analysis showed orphaned tool calls without corresponding results
SELECT role, parts FROM messages WHERE session_id = '...' ORDER BY created_at;
-- Result: assistant message with tool_call, followed by user message (missing tool result)
```

The Anthropic API strictly requires that every  tool_use  block must be immediately followed by a
 tool_result  block. When conversations were resumed after the outage, the API returned:

```
POST "https://api.anthropic.com/v1/messages": 400 Bad Request
{"error":{"message":"messages.92: `tool_use` ids were found without `tool_result` blocks
immediately after: toolu_01AWFE1DffRECo8rXkvRUkxx"}}
```

### The Fix

1. Message history validation:  convertMessages()  now tracks orphaned tool calls and injects
placeholder error results, allowing the conversation to continue
2. Error propagation: Ensures all error paths in  stream()  and  send()  methods properly emit  
EventError  events, preventing UI freezes

#### More specifically
- Scanning for tool_use blocks without corresponding tool_result blocks
- Automatically inserting missing tool_result blocks with appropriate error messages
- Improving error propagation to preserve original API errors rather than wrapping them
- Adding comprehensive test coverage for these scenarios

This ensures that conversations can continue even when tool calls are interrupted,
preventing users from getting stuck and having to start new conversations.

### Related issue/discussion: N/A

### Checklist before requesting a review

- [x] I have read [`CONTRIBUTING.md`](https://github.com/charmbracelet/.github/blob/main/CONTRIBUTING.md)
- [x] I have performed a self-review of my code

mpj committed:
fix(anthropic): handle orphaned tool calls and improve error propagation

- Add detection and handling of orphaned tool_use blocks in message history
- Automatically insert missing tool_result blocks with appropriate error messages
- Improve error propagation to preserve original API errors
- Add comprehensive test coverage for orphaned tool call scenarios
- Fix test failures by correcting createMockError function

This prevents the "tool_use ids were found without tool_result blocks" error
that can occur when the assistant's response is interrupted or when there
are issues with tool execution.

charmcli commented:
All contributors have signed the CLA  ✍️ ✅<br/><sub>Posted by the ****CLA Assistant Lite bot****.</sub>

mpj committed:
fix(anthropic): merge partial tool_result with placeholders and guard nil Response in retries

- When next message is Tool but missing some tool_result blocks, merge real results with placeholder error results for missing tool_use IDs into a single user message right after assistant
- Skip processing of the original Tool message to avoid duplicate tool_result blocks
- Add tests covering partial coverage merge behavior and nil Response handling
- Defensive nil checks around anthropic.Error.Response to prevent panics when reading headers or building error strings

💘 Generated with Crush
Co-Authored-By: Crush <crush@charm.land>

taciturnaxolotl commented:
this should be fixed in #2622
